packages/ember-metal/lib/injected_property.js

```diff
@@ -21,12 +19,13 @@ import { descriptorFor } from './meta';
          to the property's name
   @private
 */
-export default function InjectedProperty(type, name) {
-  this.type = type;
-  this.name = name;
+export default class InjectedProperty extends ComputedProperty {
+  constructor(type, name) {
+    super(injectedPropertyGet);
 
-  this._super$Constructor(injectedPropertyGet);
-  AliasedPropertyPrototype.oneWay.call(this);
```

Comment:
Was this doing something which is no longer needed?

Reply:
That line was ensuring that you can clobber an injection if you set it. This is the default behaviour of getter-only, non-readonly computed properties, which we are now inheriting.

FWIW, there is a test specifically for this added in https://github.com/emberjs/ember.js/pull/9889